ivy/functional/ivy/gradients.py

```diff
@@ -257,18 +275,19 @@ def adam_step(
     return ((alpha * mw) / (vw**0.5 + epsilon)), mw, vw
 
 
+
 # Optimizer Updates #
 
 
 @to_native_arrays_and_back
 @handle_nestable
 def optimizer_update(
-    w: Union[ivy.Array, ivy.NativeArray],
-    effective_grad: Union[ivy.Array, ivy.NativeArray],
+    w: Union[ivy.Array, ivy.NativeArray, ivy.Container],
```

Comment:
we should not include `ivy.Container` in the type hints of any nestable functions.